_studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp

```diff
@@ -976,19 +978,9 @@ namespace MPEG2EncoderHW
 
         ENCODE_CAPS EncCaps = {};
 
-        mfxStatus sts = CheckHwCaps(core, par, ext, &EncCaps);
+        sts = CheckHwCaps(core, par, ext, &EncCaps);
         MFX_CHECK_STS(sts);
 
-        if (par->mfx.FrameInfo.Width > 0x1fff || (par->mfx.FrameInfo.Width & 0x0f) != 0)
-        {
-            return MFX_ERR_INVALID_VIDEO_PARAM;
-        }
-        mfxU32 mask = (par->mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_PROGRESSIVE)? 0x0f:0x1f;
```

Comment:
This code also checks alignment for width and height. Please, don't delete this check, because IsGuidSupported doesn't check alignment.